### ISSUE TITLE
Add support for easy syntax

### ISSUE THREAD

**Issue by jefft0** [IIIM-IS/AERA#22]
Currently, the strict syntax of Replicode is difficult to understand, especially since indentation is significant, For example,

    f_b_is_a_ball:(fact b_is_a_ball 0us MAX_TIME 1 1) [[SYNC_AXIOM now 1 forever stdin nil]]

is valid. But if you naturally want to split up a long line, the following is a syntax error and takes a long time to figure out why.

     f_b_is_a_ball:(fact b_is_a_ball 0us MAX_TIME 1 1)
                   [[SYNC_AXIOM now 1 forever stdin nil]]

Similarly, the following is correct syntax.

    b_is_blue:(mk.val b 
    color 
    blue_color 
    1
    ) |[]

But if you naturally want to indent, the following is invalid because the indentation means a new set of parentheses (without giving the parentheses).

    b_is_blue:(mk.val b 
       color 
       blue_color 
       1
    ) |[]

Finally, in the following example, the `[]` on the end of the first line is not an empty list, as one would assume. It is a special syntax marker which means to start a new list, controlled by the strict indentation of the following lines..

    ball:(ont 1) []
       [SYNC_ONCE now 0 forever root nil]

Solution: Update the parser so that the directive `!easy` turns on easy syntax for the code that follows in the current file (not in another file from a `!load` directive). The easy syntax has the following features.

* Indentation, extra spaces, tabs and newlines are not significant. (Newlines are only significant to end a preprocessor directive like `!def` and for the end of a comment which starts with `;` .)
* Instead of using indentation for a new set of parentheses, the parentheses are required.
* Instead of using `[]` to mean that a new list starts on the following lines, each list must start with `[` and end with `]`.
* `[]` is an empty list. (For backwards compatibility, we also allow `|[]` as an empty list.)
* The directive `!strict` turns off easy syntax and returns to the strict syntax. Therefore, if an existing code file has the strict syntax, one can insert a new section of easy syntax, surrounded by `!easy` and `!strict` .

In summary, simply putting `!easy` at the top of a code file allows the easy syntax which is more intuitive, especially for non-experts.
